demo/network/src/lib.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright 2017 Parity Technologies (UK) Ltd.
+// This file is part of Substrate Demo.
+
+// Substrate Demo is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate Demo is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate Demo.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Substrate Demo-specific network implementation.
+//!
+//! This manages gossip of consensus messages for BFT and for parachain statements,
+//! parachain block and extrinsic data fetching, communication between collators and validators,
+//! and more.
+
+extern crate substrate_bft as bft;
+extern crate substrate_network;
+extern crate substrate_primitives;
+
+extern crate demo_api;
+extern crate demo_consensus;
+extern crate demo_primitives;
+
+extern crate ed25519;
+extern crate futures;
+extern crate parking_lot;
+extern crate tokio;
+extern crate rhododendron;
+
+#[macro_use]
+extern crate log;
+
+pub mod consensus;
+
+use demo_primitives::{Block, Hash, Header};
+use substrate_network::{NodeIndex, Context, Severity};
+use substrate_network::consensus_gossip::ConsensusGossip;
+use substrate_network::{message, generic_message};
+use substrate_network::specialization::Specialization;
+use substrate_network::StatusMessage as GenericFullStatus;
+
+/// Demo protocol id.
+pub const PROTOCOL_ID: ::substrate_network::ProtocolId = *b"dot";
```

Comment:
change this?

Reply:
I've left it as is for now so that `substrate-demo` could sync to the Krumme Lanke network. I'll make it configurable through the spec file in the following PR.